### PR TITLE
fix: cap long prefix paths

### LINKED ISSUE #198: riot generates paths longer than the FS can support
If a virtual environment has many dependencies (the odds for "fat" venvs created with `create=True` are in fact higher), then the generated path could be longer than the FS can support and the venv creation fails. In such cases it's perhaps best replacing the path containing the package spec with the venv hash.

### PR DESCRIPTION
**Pull request by P403n1x87**
This change ensures that riot does not generate prefix paths that are too long and might go beyond the limits imposed by the FS.

Fixes #198.

### CODE REVIEW

**Comment by majorgreys** [comment]
Is it necessary to drop 3.6 support here?

**Comment by P403n1x87** [comment]
> Is it necessary to drop 3.6 support here?

Not here, I can make another PR for that, but CI couldn't pull 3.6 anymore and I thought it didn't make sense to invest time in trying to fix it for such an old version.

**Comment by ZStriker19** [comment]
What's FS?

**Review by ZStriker19 — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Comment by mabdinur** [comment]
macos-latest 3.6 and ubuntu-latest 3.6 are required actions. Can we remove this check?

**Comment by P403n1x87** [comment]
> What's FS?

File System 🙂

**Comment by P403n1x87** [comment]
> macos-latest 3.6 and ubuntu-latest 3.6 are required actions. Should we remove this check to unblock this PR?

Yes, I think we should. But I would like to drop support for 3.6 in another PR first. I'll try to open it today 🤞

File: tests/test_unit.py [hunk 1]
```diff
@@ -9,7 +9,6 @@
 from riot.riot import Interpreter, run_cmd, Session, Venv, VenvInstance
 from tests.test_cli import DATA_DIR
```
Comment (mabdinur) [inline tests/test_unit.py:11]:
```suggestion


```

**Review by mabdinur — APPROVED** [review]

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**